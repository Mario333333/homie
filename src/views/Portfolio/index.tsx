import PortfolioItem from "../../components/PortfolioItem";
import useAxios from "../../hooks/useAxios";
import { PortfolioData } from "../../utils/ts/componentTypes";

const Portfolio = () => {
  const { data, error } = useAxios<Array<PortfolioData>>("briefcase");
  console.log(data);

  return (
    <div className="portfolio-container">
      {error === false &&
        data !== null &&
        data.map((item: PortfolioData, index: number) => {
          return <PortfolioItem item={item} key={`portfolio-${index}`} />;
        })}
    </div>
  );
};

export default Portfolio;
