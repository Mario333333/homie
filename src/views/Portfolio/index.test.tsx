import Portfolio from "./index";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";
import { act } from "react-dom/test-utils";
const history = createMemoryHistory();

describe("Portfolio", () => {
  afterEach(cleanup);

  test("portfolio snapshot", () => {
    const { asFragment } = render(<Portfolio />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("portfolio defined component and content", () => {
    act(() => {
      history.push("/portfolio");
      const component = render(
        <Router history={history}>
          <Switch>
            <Route exact path="/portfolio">
              <Portfolio />
            </Route>
          </Switch>
        </Router>
      );
      expect(component).toBeDefined();
      component.debug();
    });
  });
});
