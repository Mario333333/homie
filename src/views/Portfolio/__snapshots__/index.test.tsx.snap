// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio portfolio snapshot 1`] = `
<DocumentFragment>
  <div
    class="portfolio-container"
  />
</DocumentFragment>
`;
